app/bundles/EmailBundle/Form/Type/DashboardEmailsInTimeWidgetType.php

```diff
@@ -40,6 +86,50 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'required'   => false,
             ]
         );
+        $user             = $this->userHelper->getUser();
+        $companies        = $this->companyRepository->getCompanies($user);
+        $companiesChoises = [];
+        foreach ($companies as $company) {
+            $companiesChoises[$company['id']] = $company['companyname'];
+        }
+        $builder->add('companyId', 'choice', [
+            'label'      => 'mautic.email.companyId.filter',
+            'choices'    => $companiesChoises,
+            'label_attr' => ['class' => 'control-label'],
+            'attr'       => ['class' => 'form-control'],
+            'empty_data' => '',
+            'required'   => false,
+        ]);
+        /** @var Campaign[] $campaigns */
+        $campaigns        = $this->campaignRepository->findAll();
+        $campaignsChoices = [];
+        foreach ($campaigns as $campaign) {
+            $campaignsChoices[$campaign->getId()] = $campaign->getName();
+        }
+        $builder->add('campaignId', 'choice', [
+                'label'      => 'mautic.email.campaignId.filter',
+                'choices'    => $campaignsChoices,
+                'label_attr' => ['class' => 'control-label'],
+                'attr'       => ['class' => 'form-control'],
+                'empty_data' => '',
+                'required'   => false,
+            ]
+        );
+        /** @var LeadList[] $segments */
+        $segments        = $this->segmentsRepository->getLists($user);
+        $segmentsChoices = [];
+        foreach ($segments as $segment) {
+            $segmentsChoices[$segment->getId()] = $segment->getName();
```

Comment:
I'm getting `Call to a member function getId() on array` on this line when trying to edit the Emails In Time widget in the dashboard. Is it really working for you?